substrate/frame/assets/src/functions.rs

```diff
@@ -132,6 +132,9 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Some(details) => details,
 			None => return DepositConsequence::UnknownAsset,
 		};
+		if details.status == AssetStatus::Destroying {
```

Comment:
nit: I'd use `matches!` instead of `==`. Same for usage below.

Reply:
what's the benefit? with `==` I follow the style within the function

Reply:
Pattern matching is the standard way of comparing `enum`s and it doesn't require `PartialEq`/`Eq`, but in this case it's not a big deal.